### PR TITLE
New dataset

### PR DESCRIPTION
**Pull request by johannes-b97**
Choose between two datasets in prepare, implemented splittng in test/train if necessary. should work with dvc

### CODE REVIEW

File: src/prepare.py [hunk 1]
```diff
@@ -75,31 +76,49 @@ def review_sentences(review, tokenizer, remove_stopwords=False):
...
     output = sys.argv[2]
-
-    input_train = os.path.join(input, 'trainset.zip')
-    input_test = os.path.join(input, 'testset.zip')
-    print(input_train, input_test)
+    name = sys.argv[3]
+    if len(sys.argv) == 5:
+        name2 = sys.argv[4]        
+        input_train = os.path.join(input, name)
+        input_test = os.path.join(input, name2)
```
Comment (DennisHamann) [inline src/prepare.py:83]: Kann es hier nicht passieren, dass Test und Train aus unterschiedlichen Sets geladen werden?

File: src/prepare.py [hunk 2]
```diff
@@ -119,7 +138,7 @@ def review_sentences(review, tokenizer, remove_stopwords=False):
     model.init_sims(replace=True)
 
     # Saving the model for later use. Can be loaded using Word2Vec.load()
-    model_name = "features_model"
+    model_name = f"{num_features}features_model"
```
Comment (DennisHamann) [inline src/prepare.py:141]: Alle Dateinamen sollen unabhängig von num_features bleiben.

File: src/prepare.py [hunk 3]
```diff
@@ -75,31 +76,49 @@ def review_sentences(review, tokenizer, remove_stopwords=False):
...
+        zf.close()  
+        df = pd.read_csv('data/prepared/dataset.csv')
+        df_new = df[df["Text"].notnull()]
+        train, test = train_test_split(df_new, test_size=split)  
+        path_train = os.path.join(output, 'Train.csv')
+        path_test = os.path.join(output, 'Test.csv')
+        train.to_csv(path_train)   
+        test.to_csv(path_test)   
+        train = pd.read_csv('data/prepared/Train.csv')
+        test = pd.read_csv('data/prepared/Test.csv')
```
Comment (DennisHamann) [inline src/prepare.py:112]: test/train variablen statisch und nicht abhängig von input

File: src/prepare.py [hunk 4]
```diff
@@ -64,7 +65,7 @@ def review_sentences(review, tokenizer, remove_stopwords=False):
 
 
 if __name__ == '__main__':
-    if len(sys.argv) != 3:
+    if len(sys.argv) != 4 and len(sys.argv) != 5:
```
Comment (DennisHamann) [inline src/prepare.py:68]: Wieso werden hier 4 oder 5 Argumente erwartet und was soll da drin stehen?
Bitte dann auch in dvc ergänzen, dass die Befehle durchlaufen
(dvc pipeline show evaluate.dvc --ascii -c)

Comment (DennisHamann) [reply src/prepare.py:68]: am besten wird kann das Skript so ausgeführt werden:
python3 src/prepare.py data/dataset data/prepared  (dann wird default imbd genommen)
python3 src/prepare.py data/dataset data/prepared --dataset=clothes
python3 src/prepare.py data/dataset data/prepared --dataset=imbd

File: src/prepare.py [hunk 5]
```diff
@@ -75,31 +76,49 @@ def review_sentences(review, tokenizer, remove_stopwords=False):
...
+        path_test = os.path.join(output, 'Test.csv')
+        train.to_csv(path_train)   
+        test.to_csv(path_test)   
+        train = pd.read_csv('data/prepared/Train.csv')
+        test = pd.read_csv('data/prepared/Test.csv')
     
     sentences = []
     print("Parsing sentences from training set")
-    for review in train["text"]:
+    for review in train["Text"]:
```
Comment (DennisHamann) [inline src/prepare.py:116]: für das alte Datenset gibt es einen Keyerror. Es muss für beide funktionieren

File: src/prepare.py (hunk 1 above)
Comment (johannes-b97) [reply src/prepare.py:83]: Moeglich ja, aber mache Datensets liegen von vornerein als 2 Zip.files vor, also muss ich ja zwei Adressen uebergeben

File: src/prepare.py (hunk 2 above)
Comment (johannes-b97) [reply src/prepare.py:141]: Stimmt!

File: src/prepare.py (hunk 3 above)
Comment (johannes-b97) [reply src/prepare.py:112]: Ja das kann eigentlich raus, die habe ich davor ja gerade dort abgespeichert in 107-110

File: src/prepare.py [hunk 6]
```diff
@@ -138,7 +136,7 @@ def review_sentences(review, tokenizer, remove_stopwords=False):
     model.init_sims(replace=True)
 
     # Saving the model for later use. Can be loaded using Word2Vec.load()
-    model_name = f"{num_features}features_model"
+    model_name = "w2v_model"
```
Comment (DennisHamann) [inline src/prepare.py:139]: Du kannst nicht einfach model_name ändern, da es Abhängikeiten in featurization.py gibt.

File: src/prepare.py [hunk 7]
```diff
@@ -102,18 +102,16 @@ def review_sentences(review, tokenizer, remove_stopwords=False):
...
         train.to_csv(path_train)   
         test.to_csv(path_test)   
-        train = pd.read_csv('data/prepared/Train.csv')
-        test = pd.read_csv('data/prepared/Test.csv')
-    
+        
     sentences = []
     print("Parsing sentences from training set")
-    for review in train["Text"]:
+    for review in train["text"]:
```
Comment (DennisHamann) [inline src/prepare.py:114]: mit 'text' klein geschrieben funktioniert es nicht mehr für 
python3 src/prepare.py data/dataset/ data/prepared/ clothing_set.zip

Es muss für beide funktionieren.

File: src/prepare.py [hunk 8]
```diff
@@ -102,18 +102,16 @@ def review_sentences(review, tokenizer, remove_stopwords=False):
         zf.extractall(output)
         zf.close()  
         df = pd.read_csv('data/prepared/dataset.csv')
-        df_new = df[df["Text"].notnull()]
+        df_new = df[df["text"].notnull()]
```
Comment (DennisHamann) [inline src/prepare.py:105]: siehe Kommentar unten

File: src/prepare.py (hunk 7 above)
Comment (DennisHamann) [reply src/prepare.py:114]: Weiterhin unterscheiden sich die Formate der csv Dateien. Hier musst du eine Fallunterscheidung vornehmen.

format imbd csv:
text,label
"text...",0

format clothes csv:
,Unnamed: 0,Clothing ID,Age,Title,Text,Rating,Recommended IND,Positive Feedback Count,Division Name,Department Name,Class Name,Unnamed: 11,Unnamed: 12,Unnamed: 13,Unnamed: 14,Unnamed: 15,Unnamed: 16,Unnamed: 17,Unnamed: 18
21987,21987,860,37,"Cute, flattering, and goes with everything","I bought this top to add some more color and pattern to my summer wardrobe. it is really cute and a flattering cut. i just had a baby, so it's perfect for showing some curves without clinging. if anything, i'd say it maximizes my bustline more than i expected, but is still very flattering. i think i'm still getting used to my new bustline ;) soft, comfy fabric. not too hot. i get compliments when i wear it!",5,1,0,General,Tops,Knits,,,,,,,,

Bitte die Formate angleichen.

Comment (DennisHamann) [reply src/prepare.py:114]: Sodass in data/prepared die Dateien Test.csv und Train.csv existieren mit dem Format

text,label
"some text",0
"some other text",1
